test/com/google/enterprise/adaptor/DocIdSenderTest.java

```diff
@@ -491,6 +564,48 @@ public void testAclItemToString() {
     assertEquals(golden, "" + new DocIdSender.AclItem(id, acl));
   }
 
+  private static Map<GroupPrincipal, Collection<Principal>> sampleGroups() {
+    return ImmutableMap.<GroupPrincipal, Collection<Principal>>of(
+        new GroupPrincipal("g1"), ImmutableList.<Principal>of(
+            new UserPrincipal("u1"), new GroupPrincipal("g2")),
+        new GroupPrincipal("g2"), ImmutableList.<Principal>of(
+            new UserPrincipal("u2"), new GroupPrincipal("g3")),
+        new GroupPrincipal("g3"), ImmutableList.<Principal>of(
+            new UserPrincipal("u3"), new GroupPrincipal("g4")));
+  }
+
+  private static Map<GroupPrincipal, Collection<Principal>> emptyGroups() {
+    return ImmutableMap.<GroupPrincipal, Collection<Principal>>of();
+  }
+
+  // TODO(bmj): Uncomment @SafeVarargs when moving to Java 7 and remove the
+  // @SuppressWarnings("unchecked") usages above.
+  // @SafeVarargs
+  private static List<List<Map.Entry<GroupPrincipal, Collection<Principal>>>>
+      expectedResult(int batchSize,
```

Comment:
Could you document what this does? And write a unit test for it? I presume it takes the group definitions as inputs and generates the expected batches recorded by the mock?